### PR TITLE
feat(BUX-566): metrics

### PR DESCRIPTION
**Pull request by chris-4chain**
# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/BuxOrg/bux-server/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/BuxOrg/bux-server/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

This PR is tightly coupled with the one from `bux` repository - they should be tested together. 

Generally, I introduced metrics collection in `bux` package and initialized it within `bux-server`. 
> NOTE: Additional metrics of incoming requests will be implemented when transition to `gin` is complete. 

The metrics that prometheus can collect are available at the `/metrics` endpoint - which is a standard approach. 

If you'd like to test it locally with local Prometheus & Grafana - contact me for help with configuration and example dashboards. 
It can also be tested manually by checking the content od `/metrics` endpoint. 

Example dashboard:
Zrzut ekranu 2024-02-09 144241

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/BuxOrg/bux-server
 go 1.21.5
 
 require (
-	github.com/BuxOrg/bux v0.14.3-0.20240201141930-5ac2858facdc
+	github.com/BuxOrg/bux v0.14.4-0.20240212060332-36a09b80059f
```
Comment (chris-4chain) [inline go.mod:6]: I used a bux branch [feat-567-metrics](https://github.com/BuxOrg/bux/tree/feat-567-metrics) as a temporary dependency. 
*Remember* to switch it after this branch is merged to `rc-v1.1`.

File: metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// Metrics is the metrics collector
+type Metrics struct {
+	gatherer   prometheus.Gatherer
+	registerer prometheus.Registerer
+}
+
+// newMetrics will create a new Metrics object; this is private to ensure only one, global instance is created
```
Comment (wregulski) [inline metrics/metrics.go:16]: I think that this kind of comment is unnecessary - for me we should keep the rule from our code standards:
"Write self-documenting code and minimize comments."
I think that this function is self-explanatory enough and doesn't need a comment.

**Review by wregulski — APPROVED** [review]

File: metrics/metrics.go (hunk 2 above)
Comment (chris-4chain) [reply metrics/metrics.go:16]: I've intentionally added this comment to note why the "constructor" is private and mention the `global` instance. Maybe only the first part `newMetrics will create a new Metrics object;` is unnecessary. What do you think?

**Review by Nazarii-4chain — APPROVED** [review]

File: metrics/metrics.go (hunk 2 above)
Comment (wregulski) [reply metrics/metrics.go:16]: Okay, I agree we can leave the part after ;

Comment (chris-4chain) [reply metrics/metrics.go]: Done 👍 - check the new comment.

**Review by dorzepowski — APPROVED** [review]

**Comment by chris-4chain** [comment]
I switched `bux` dependency to this commit on `rc-v1.1`:
https://github.com/BuxOrg/bux/commit/b0a6e9fa55c4d9d702aa923e00ac1c9a925a5da4

**Pull request merged**